backend/src/db/models/User.js

```diff
@@ -72,7 +72,7 @@ export default (connection, mongoose) => {
      * @returns {Promise<boolean>}
     */
     async isEmailConfirmed(email) {
-      const user = await User.findOne({ where: { email } });
+      const user = await User.scope('withEmailToken').findOne({ where: { email } });
       return user.mailToken === null;
     }
```

Comment:
Dans `security.js`, dans la route de login. On a deux facon de faire pour vérifier le mot de passe et l'email.

Pour le mot de passe, on va demander le mot de passe chifré via un scope mis dans le `userService.findLogin()` et on lui passe le password non hashé (recupérer du body) dans la method `checkPassword`. Cette method va utiliser `this` (ligne 66 de ce fichier) pour allez chercher le mot de passe dans l'instance du model (possible car on a passer un scope pour lui demander de recupérer le hash).

Or, pour l'email, `userService.findLogin()` utilise le scope `withoutMailToken`. Il a donc besoin de refaire un find sur la base de donnée pour avoir accès à l'email token.

Tu devrais unifier les deux comportement. Pour moi `userService.findLogin()` devrait récuprérer aussi le `mailToken` devrait utiliser le `this.emailToken` pour vérifier si l'email est confirmé.

```suggestion
    async isEmailConfirmed() {
      return this.mailToken === null;
    }
```